lldb/tools/lldb-dap/Protocol/ProtocolTypes.h

```diff
@@ -317,6 +317,8 @@ struct Source {
   std::optional<PresentationHint> presentationHint;
 
   // unsupported keys: origin, sources, adapterData, checksums
+
+  bool IsAssemblySource() const { return sourceReference.value_or(0) != 0; }
```

Comment:
Should we move this to a ProtocolHelpers file? I've been thinking of consolidating the protocol builders from `JSONUtils.h` and `LLDBUtils.h` into their own `ProtocolUtils.h` or `ProtocolHelpers.h`

Reply:
:100: